web/src/components/areagraph.js

```diff
@@ -70,6 +74,28 @@ AreaGraph.prototype.data = function (arg) {
 
     var that = this;
 
+    // Max total value
+    var maxTotalValue = d3.max(arg, function(d) {
+        if (!that._displayByEmissions) {
+            // in MW
+            return (d.totalProduction + d.totalImport + d.totalDischarge);
+        } else {
+            // in tCO2eq/min
+            return (d.totalCo2Production + d.totalCo2Import + d.totalCo2Discharge) / 1e6 / 60.0;
+        }
+    })
+
+    var formattingFactor = 1;
+    if (!that._displayByEmissions) {
+        // Display in GW or MW
+        formattingFactor = maxTotalValue > 1e3 ? 1e3 : 1;
```

Comment:
Instead I suggest using this:
https://github.com/tmrowco/electricitymap-contrib/blob/master/web/src/helpers/formatting.js#L13

Reply:
I tried to use d3.format before, but failed. Output of that function is a formatted string, which would not work here since we need to align the total dataset dynamically accordingly. Thats why this threshold is implemented.

Reply:
What do you mean exactly by "align the total dataset dynamically accordingly"? I'm trying to figure out if we can find a more elegant solution because some areas have `kW` scale data.

Reply:
I see. Sure 👍 . Added a new scalePower formatter.

Reply:
I just meant that data and domain need to be scaled.